flutter-for-react-native/index.md

```diff
@@ -0,0 +1,2411 @@
+                  :(new Text('Stop Blinking'))
+                )
+              )
+            )
+          ],
+        ),
+      ),
+    );
+  }
+}
+```
+
+### What are the StatefulWidget and StatelessWidget best practices?
+
+Here are a few things to consider when designing your widget.
+
+#### 1. Determine whether a widget should be a StatefulWidget or a StatelessWidget
+
+
+In Flutter, widgets are either Stateful or Stateless—depending on whether
+ they depend on a state change.
+* If a widget changes—the user interacts with it or a data feed interrupts
+ the UI, then it’s Stateful.
+
+* If a widget is final or immutable, then it's Stateless.
+
+#### 2. Determine which object manages the widget’s state (for a StatefulWidget)
+
+In Flutter, there are three primary ways to manage state:
+* The widget manages its own state
+* The parent widget manages the widget’s state
+* A mix-and-match approach
+
+When deciding which approach to use, consider the following principles:
+* If the state in question is user data, for example the checked or unchecked
+ mode of a checkbox, or the position of a slider, then the state is best managed
+  by the parent widget.
+* If the state in question is aesthetic, for example an animation, then the
+ widget itself best manages the state.
+* When in doubt, let the parent widget manage the child widget's state.
+
+#### 3. Subclass StatefulWidget and State
+
+The `MyStatefulWidget` class manages its own state—it extends
+`StatefulWidget`, it overrides the `createState()` method to create the State
+object, and the framework calls `createState()` to build the widget. In this
+example, `createState()` creates an instance of `_MyStatefulWidgetState`, which
+is implemented in the next best practice.
+
+<!-- skip -->
+```Dart
+class MyStatefulWidget extends StatefulWidget {
+  MyStatefulWidget({Key key, this.title}) : super(key: key);
+  final String title;
+
+  @override
+  _MyStatefulWidgetState createState() => new _MyStatefulWidgetState();
+}
+
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+
+  @override
+  Widget build(BuildContext context) {
+    ...
+  }
+}
+```
+
+#### 4. Add the StatefulWidget into the widget tree
+
+Add your custom `StatefulWidget` to the widget tree in the app’s build method.
+
+<!-- skip -->
+```Dart
+class MyStatelessWidget extends StatelessWidget {
+  // This widget is the root of your application.
+
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Flutter Demo',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyStatefulWidget(title: 'State Change Demo'),
+    );
+  }
+}
+```
+
+
+
+|Android|iOS|
+|:---:|:--:|
+|<img src="images/state_change_android.gif?raw=true" style="width:300px;" alt="Loading">|<img src="images/state_change_iOS.gif?raw=true" style="width:300px;" alt="Loading">|
+
+<br>
+
+In Flutter, you assign a local variable or function marked `final` with the
```

Comment:
This is a big Oopsie. We lost a header: Props. I couldn't figure out why this sentece was sitting here, or what the next screenshots were about, until I looked at the original. Please reinstate the Props heading, the (short) paragraph, and the code that follows.

Reply:
Done